medium/__init__.py

```diff
@@ -7,84 +7,139 @@
+    def _get_tokens(self, **kwargs):
+        return _request('POST', '/tokens', self.access_token, **kwargs)
 
     def exchange_authorization_code(self, code, redirect_url):
-        """Exchange the authorization code for a long-lived access token, and
-        set the token on the current Client.
-
-        :param str code: The code supplied to the redirect URL after a user
-            authorizes the application
-        :param str redirect_url: The same redirect URL used for authorizing
-            the application
-        :returns: A dictionary with the new authorizations ::
-            {
-                'token_type': 'Bearer',
-                'access_token': '...',
-                'expires_at': 1449441560773,
-                'refresh_token': '...',
-                'scope': ['basicProfile', 'publishPost']
-            }
         """
-        data = {
-            "code": code,
+        Exchange the authorization code for a long-lived access token, and set
+        both the access token and refresh on the current Client.
+
+        :param str code:
+            The code supplied to the redirect URL after a user authorizes the
+            application.
+        :param str redirect_url:
+            The same redirect URL used for authorizing the application.
+        """
+        tokens = self._get_tokens(form_data={
             "client_id": self.application_id,
             "client_secret": self.application_secret,
+            "code": code,
             "grant_type": "authorization_code",
             "redirect_uri": redirect_url,
-        }
-        return self._request_and_set_auth_code(data)
-
-    def exchange_refresh_token(self, refresh_token):
-        """Exchange the supplied refresh token for a new access token, and
-        set the token on the current Client.
+        })
+        self.access_token = tokens['access_token']
+        self.refresh_token = tokens['refresh_token']
 
-        :param str refresh_token: The refresh token, as provided by
-            ``exchange_authorization_code()``
-        :returns: A dictionary with the new authorizations ::
-            {
-                'token_type': 'Bearer',
-                'access_token': '...',
-                'expires_at': 1449441560773,
-                'refresh_token': '...',
-                'scope': ['basicProfile', 'publishPost']
-            }
+    def exchange_refresh_token(self):
+        """
+        Exchange the supplied refresh token for a new access token, and set the
+        token on the current Client.
         """
-        data = {
-            "refresh_token": refresh_token,
+        self.access_token = self._get_tokens(form_data={
             "client_id": self.application_id,
             "client_secret": self.application_secret,
             "grant_type": "refresh_token",
-        }
-        return self._request_and_set_auth_code(data)
+            "refresh_token": self.refresh_token,
+        })['access_token']
```

Comment:
Yeah, good call. I don't like the idea of assigning anything to a variable named `data`, but assigning to `tokens` would make this simpler.